File: src/main/java/com/falsepattern/zigbrains/zon/ZonFileType.java
package com.falsepattern.zigbrains.zon;

import com.falsepattern.zigbrains.common.Icons;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

import javax.swing.Icon;

public class ZonFileType extends LanguageFileType {
    @SuppressWarnings("unused") //Used by plugin.xml
    public static final ZonFileType INSTANCE = new ZonFileType();

    private ZonFileType() {
        super(ZonLanguage.INSTANCE);
    }

    @Override
    public @NonNls @NotNull String getName() {
        return "ZON File";
    }

    @Override
    public @NotNull String getDescription() {
        return "Zig object notation file";
    }

    @Override
    public @NotNull String getDefaultExtension() {
        return "zon";
    }

    @Override
    public Icon getIcon() {
        return Icons.ZON;
    }
}

File: modules/zon/src/main/kotlin/com/falsepattern/zigbrains/zon/ZonFileType.kt
package com.falsepattern.zigbrains.zon

import com.intellij.openapi.fileTypes.LanguageFileType

object ZonFileType: LanguageFileType(ZonLanguage) {
    override fun getName() = "Zon File"

    override fun getDescription() = "Zig object notation file"

    override fun getDefaultExtension() = "zon"

    override fun getIcon() = Icons.ZON
}